storage/innobase/row/row0query.cc

```
@@ -0,0 +1,634 @@
+                                           sec_index->n_core_fields,
+                                           ULINT_UNDEFINED, &m_heap);
+    /* Check if secondary record matches our search criteria */
+    RecordCompareAction action= callback.compare_record(search_tuple, sec_rec,
+                                                        sec_index, sec_offsets);
+    if (action == RecordCompareAction::PROCESS)
+    {
+      /* Lookup clustered record and process it */
+      if (!lookup_clustered_record(table, sec_index, clust_index,
+                                   sec_rec, callback, match_count))
+        break;
+    }
+    else if (action == RecordCompareAction::STOP)
+      break;
+    if (!btr_pcur_move_to_next(&m_pcur, &m_mtr)) break;
+  }
+  m_mtr.commit();
+  return (match_count > 0 || !search_tuple) ? DB_SUCCESS : DB_RECORD_NOT_FOUND;
+}
+
+bool QueryExecutor::lookup_clustered_record(dict_table_t *table,
+                                            dict_index_t *sec_index,
+                                            dict_index_t *clust_index,
+                                            const rec_t *sec_rec,
+                                            RecordCallback& callback,
+                                            ulint& match_count) noexcept
+{
+  /* Extract primary key from secondary index record */
+  dtuple_t *clust_tuple= row_build_row_ref(ROW_COPY_DATA, sec_index,
+                                           sec_rec, m_heap);
+  if (!clust_tuple) return false;
+  /* Now lookup the complete row using clustered index */
+  btr_pcur_t clust_pcur;
+  clust_pcur.btr_cur.page_cur.index= clust_index;
+
+  mtr_t clust_mtr{m_trx};
+  clust_mtr.start();
+  bool continue_processing= true;
+  bool record_processed= false;
+  dberr_t clust_err= btr_pcur_open(clust_tuple, PAGE_CUR_LE,
+                                   BTR_SEARCH_LEAF, &clust_pcur,
+                                   &clust_mtr);
```

We are discarding any error code here. If I understood it correctly, we would return `true` if an error occurs. If the function comment `@return true if should continue processing, false to break` is valid, this could lead to an infinite loop on corruption.